### PR TITLE
Fix up e2e script

### LINKED ISSUE #405: Travis: use gimme instead of building go
Currently the build scripts clone and build go 1.11beta3 from source. This is available in Travis so we should use that instead of downloading and compiling during a build.

I'll take this one.

### PR DESCRIPTION
**Pull request by robjloranger**
+ use correct variable name for module support, `GO111MODULE`
+ don't set `GO111MODULE` until after running `buffalo dev`, see #412
+ fixed old path for test repo in e2e, now `happy-path`
+ set go version in travis
+ remove building from source and artifacts

Bonus: builds should now average ~2min faster

Closes #405

### CODE REVIEW

File: scripts/get_dev_tools.sh [hunk 1]
```diff
@@ -7,15 +7,4 @@ set -xeuo pipefail
 go get github.com/golang/lint/golint
 go get github.com/golang/dep/cmd/dep
 
-GO_VERSION="go1.11beta3"
```
Comment (michalpristas) [inline scripts/get_dev_tools.sh]: what i liked on this is that this allowed me to use the stable version of Go on my dev env. while new way requires me to stay up to date with latest release

Comment (marwan-at-work) [reply scripts/get_dev_tools.sh]: @michalpristas you can still have stable/unstable versions in your dev env. This is just a set up script, so I'm not sure how you're using it. In any case, we can potentially separate the CI set up script from the dev set up script if it makes sense

**Comment by robjloranger** [comment]
I should actually also include an update to the docs about installing go for development. Or #418 would do this for me.

**Comment by robjloranger** [comment]
Agreed, #415 can make use of that after this is in

File: scripts/get_dev_tools.sh (hunk 1 above)
Comment (robjloranger) [reply scripts/get_dev_tools.sh]: I think splitting up the scripts for dev vs CI makes sense. Maybe during #397

Comment (arschles) [reply scripts/get_dev_tools.sh]: Ref this thread: https://github.com/gomods/athens/pull/418#discussion_r208064390

**Pull request merged**